src/Microsoft.VisualStudio.Services.Agent/ConfigurationStore.cs

```diff
@@ -38,14 +38,17 @@ public sealed class AgentSettings
         [DataMember(EmitDefaultValue = false)]
         public int AgentId { get; set; }
 
+        [DataMember(EmitDefaultValue = false)]
+        public string AgentCloudId { get; set; }
```

Comment:
Are we sure that this won't be null only if agent pool is hosted?

Reply:
Personally it amazes me that Azure DevOps doesn't have a `System.IsHosted` variable (tells whether the agent is connecting to Azure DevOps Service) to identify hosted agents `System.IsHostedAgent` would be very clear and also useful to others who have to make choices on whether the job is running on the hosted agent.

This Id check feels like another workaround and something the agent code itself just shouldn't know.

Reply:
@anatolybolshakov Yes, this id is presented for the agents that are provisioned by the pool provider (hosted agents), and null here will mean that it is the self-managed pool.

Reply:
@jessehouwing There is a property in the [agent pools API](https://docs.microsoft.com/en-us/rest/api/azure/devops/distributedtask/pools/get?view=azure-devops-rest-6.0) that shows that the agent is hosted by MS. The problem is that this property is showing that the agent belongs to the ms-hosted pool, while the previous variant of the check in-fact checked that the agent is managed by a pool provider (which is not mean that the agent is ms-hosted), verification that is based on the check of AgentCloudId property ideologically closer to the one that was previously. I think in this particular case there is not a very good naming of this _IsHosted_ property on the agent side that could confuse.

Reply:
+1 oe the confusing names.

I've seen the request for a variable or condition to check tor hosted many times though. Both in forums and stackoverflow.

Reply:
@jessehouwing for a new variable to determine host environment - I would suggest to open  https://developercommunity.visualstudio.com/search?space=21 to get right eyes on this as feature request - since this would be more general change not only related to the agent side.